MantidQt/CustomInterfaces/src/Reflectometry/ReflDataProcessorPresenter.cpp

```diff
@@ -392,34 +464,53 @@ std::string ReflDataProcessorPresenter::takeSlice(const std::string &runNo,
 /** Plots any currently selected rows */
 void ReflDataProcessorPresenter::plotRow() {
 
-  // if uniform slicing is empty plot normally
-  std::string timeSlicing = m_mainPresenter->getTimeSlicingOptions();
-  if (timeSlicing.empty()) {
+  const auto items = m_manager->selectedData();
+  if (items.size() == 0)
+    return;
+
+  // If slicing values are empty plot normally
+  std::string timeSlicingValues = m_mainPresenter->getTimeSlicingValues();
+  if (timeSlicingValues.empty()) {
     GenericDataProcessorPresenter::plotRow();
     return;
   }
 
-  // Parse time slices
   std::vector<double> startTimes, stopTimes;
-  parseTimeSlicing(timeSlicing, startTimes, stopTimes);
+  std::string timeSlicingType = m_mainPresenter->getTimeSlicingType();
+
+  // Num of slices can be predetermined with uniform even or custom slicing
+  if (timeSlicingType == "UniformEven") {
+    const auto &row = items.at(0).at(0);
+    const std::string wsName = getReducedWorkspaceName(row, "TOF_");
+    parseUniform(timeSlicingValues, timeSlicingType, wsName, startTimes,
+                 stopTimes);
+  } else if (timeSlicingType == "Custom") {
+    parseCustom(timeSlicingValues, startTimes, stopTimes);
+  }
+
```

Comment:
After some investigation, I think we only need to store how many slices are used for each run in each group. We don't actually need the start/stop times after slicing is done.